language-plutus-core/src/Language/PlutusCore/Constant/Apply.hs

```diff
@@ -218,6 +220,13 @@ applyTypedBuiltinName
     => TypedBuiltinName a r -> a -> [Value TyName Name ()] -> QuoteT (Evaluate m) ConstAppResult
 applyTypedBuiltinName (TypedBuiltinName _ schema) = applyTypeSchemed schema
 
+-- we default to 'False' in accordance with the spec
+verifySignature :: BSL.ByteString -- ^ Public Key
+                -> BSL.ByteString -- ^ Message
+                -> BSL.ByteString -- ^ Signature
+                -> Bool
+verifySignature pubKey msg sig = fromMaybe False $ maybeCryptoError (verify <$> publicKey (BSL.toStrict pubKey) <*> pure (BSL.toStrict msg) <*> signature (BSL.toStrict sig))
```

Comment:
Please move this to some other module. We do not implement crypto primitives here, this module only deals with type-safe constant application. For hashes we have `Data.ByteString.Lazy.Hash`. I'd just create `Crypto.hs` and put the context of `Data.ByteString.Lazy.Hash` and your `verifySignature` there.